scripts/build/gen_bindesc_checksums.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+
+import sys
+import hashlib
+from elftools.elf.elffile import ELFFile
+from elftools.elf.constants import SH_FLAGS
+
+
+BINDESC_OFFSETOF_DATA = 4
+HASH_ALGORITHMS = {
+    "md5": hashlib.md5,
+}
+
+
+def set_checksums(filename):
+    with open(filename, 'rb+') as file_stream:
+        elffile = ELFFile(file_stream)
```

Comment:
Not so sure this is really that useful, if the image is for MCUboot then the hex and bin files will have headers and footers added, so an MD5 hash (not a checksum) is not useful here. And if an elf file is split up, e.g. into different memory regions, this also does not have much use because the MCU would not ever really be able to verify that hash.

Reply:
This is not intended as a cryptographic hash for the MCU to verify (although it can be). It's a method for creating a "build ID". The use case, as phrased by @JordanYates is:

> The GNU build ID tells you, without any doubt, the binary running on this device is the same as the binary I am inspecting on my computer.

It's harder to do with the hash given by imgtool: that only associates the binary running on a device with a binary on your computer, not an ELF, and it only works if your image is signed - it won't work on the bootloader itself or on a native_posix image. Also, if the image includes a build time, you will get a different hash for different builds.

Reply:
Ah I see what you are trying to do here, in that case let's remove "checksum" from this because it is not a checksum. Build ID is a good way to refer to it.